website/docs/sdk-reference/dotnet.md

```diff
@@ -140,12 +140,49 @@ You can close all open clients at once using the `ConfigCatClient.DisposeAll()`
 
 ```csharp
 User userObject = new User("#UNIQUE-USER-IDENTIFIER#");  // Optional User Object
-var value = client.GetValue("keyOfMySetting", false, userObject);
+var value = client.GetValue("keyOfMyFeatureFlag", false, userObject);
 ```
 
 ```csharp
 User userObject = new User("#UNIQUE-USER-IDENTIFIER#"); // Optional User Object
-var value = await client.GetValueAsync("keyOfMySetting", false, userObject);
+var value = await client.GetValueAsync("keyOfMyFeatureFlag", false, userObject);
+```
+
+:::caution
+Please keep in mind that you need to supply an argument for the `defaultValue` parameter (more precisely, the `T` generic type parameter)
+whose type corresponds to the kind of the feature flag or setting to evaluate as described by the following table.
+:::
+
+<div id="setting-type-mapping"></div>
+
+| Setting Kind   | Type parameter `T`                |
+| -------------- | --------------------------------- |
+| On/Off Toggle  | `bool` / `bool?`                  |
+| Text           | `string` / `string?`              |
+| Whole Number   | `int` / `int?` / `long` / `long?` |
+| Decimal Number | `double` / `double?`              |
+
+Besides the types above, you can also use `object` / `object?` regardless of the setting kind. This, however, is not recommended as
+it may involve [boxing](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/types/boxing-and-unboxing).
+
+Using any other type will result in `ArgumentException`.
+
+In case you specify an allowed type but it mismatches the setting kind, an error message will be logged and `defaultValue` will be returned.
+Please pay attention to this behavior when you don't explicitly specify the type argument but rely on type inference because
```

Comment:
When relying on type inference and not explicitly specifying the type argument, be mindful of potential type mismatch issues, especially with number types. For example, if you mistakenly infer the type as int instead of double when evaluating a decimal setting, like client.GetValue("keyOfMyDecimalSetting", 0), it will return the defaultValue (0) instead of the expected decimal value. 

To correctly evaluate a decimal setting, you should use: